aten/src/ATen/native/mkldnn/Conv.cpp

```diff
@@ -35,43 +35,78 @@ std::tuple<at::Tensor,at::Tensor,at::Tensor> mkldnn_convolution_backward(
 #else // AT_MKLDNN_EBABLED
 
 #include <ATen/mkldnn/Runtime.h>
+#include <ATen/native/mkldnn/MKLDNNCommon.h>
+#include <ATen/native/mkldnn/Utils.h>
 
 using namespace mkldnn;
 
 namespace at { namespace native {
 
-constexpr int input_batch_size_dim = 0;  // also grad_input
-constexpr int input_channels_dim = 1;
-constexpr int output_batch_size_dim = 0;  // also grad_output
-constexpr int output_channels_dim = 1;
-constexpr int weight_output_channels_dim = 0;
-constexpr int weight_input_channels_dim = 1;
+Tensor _ideep_conv2d(
+    const Tensor& input, const Tensor& weight, const Tensor& bias,
+    at::IntArrayRef padding, at::IntArrayRef stride, at::IntArrayRef dilation, int64_t groups) {
 
-// Often written as 2 + max_dim (extra dims for batch size and channels)
-constexpr int max_dim = 3;
+  auto stride_vec = expand_param_if_needed(stride, 2);
```

Comment:
`expand_param_if_needed` is not needed, in case we still call `mkldnn_convolution` from `_convolution()`:
1. inside `use_mkldnn` from `Convolution.cpp`, it will regulate mkldnn for 2d convolution only, and 3d convolution will go `THNN` for now. and 1d convolution reuses 2d logic. So `stride, padding, dilation` can be only of vector of 2 here.
2. in near future, when `ideep` can handle 3d convolution, `mkldnn_convolution` will be able to handle both 2d and 3d in one place, just like `cudnn_convolution` did.